### PR TITLE
[BS-205][TW] Reinstate Get BS Periods in RAML docs

### PR DESCRIPTION
**Pull request by glidester**

### CODE REVIEW

File: public/api/conf/1.0/application.raml [hunk 1]
```diff
@@ -93,103 +93,103 @@ documentation:
...
+
+              The two possible values are ‘DA2_BS_ATTENDED’ or ‘DA2_BS_UNATTENDED’.
+            type: string
+            required: true
+            example: DA2_BS_ATTENDED
+          Staff-Id:
+            description: |
+              The Staff identifier associated with the human attending the operation that generated this request.
+
+              Must be a 7 digit numeric value
```
Comment (udinobi) [inline public/api/conf/1.0/application.raml:138]: Is it worth to add that
- for `DA2_BS_ATTENDED` `Staff-id` cannot be `"0000000"`
- for `DA2_BS_UNATTENDED` `Staff-id` can only be `"0000000"`
?

Comment (glidester) [reply public/api/conf/1.0/application.raml:138]: Yes, good point. I'll add that in.

File: public/api/conf/1.0/application.raml [hunk 2]
```diff
@@ -93,103 +93,103 @@ documentation:
...
+            body:
+              application/json:
+                type: types.errorResponse
+                examples:
+                  resourceNotFound:
+                    description: The supplied NINO was not found
+                    value:
+                      code: RESOURCE_NOT_FOUND
+
+          502:
```
Comment (udinobi) [inline public/api/conf/1.0/application.raml:186]: Shouldn't 502 not added here but in a "general errors" section ?

Comment (glidester) [reply public/api/conf/1.0/application.raml:186]: 502 is not covered in the API Platform general errors reference page: https://developer.qa.tax.service.gov.uk/api-documentation/docs/reference-guide#errors

We should never return this ourselves, if we get this from IF we should return 500. Have a look at some other api project  implementations to see what they do.

File: public/api/conf/1.0/application.raml (hunk 1 above)
Comment (glidester) [reply public/api/conf/1.0/application.raml:138]: I've added:
"Please note: when the 'Request-Type' header has the value ‘DA2_BS_UNATTENDED’ then this header's value must equal '0000000' "
We don't check that for ‘DA2_BS_ATTENDED’ that value is not '0000000' do we?

Comment (udinobi) [reply public/api/conf/1.0/application.raml:138]: I'm removing the UserId header when the value is "0000000". EIS is not expecting the UserId header, and accordingly we need to remove it, for DA2_BS_UNATTENDED. If we allow "0000000" for DA2_BS_ATTENDED this header will be removed and we don't want this of course.

File: public/api/conf/1.0/application.raml (hunk 2 above)
Comment (udinobi) [reply public/api/conf/1.0/application.raml:186]: Ok. I'll remove it then from `handleUpstreamError` and as `Error` enum.

**Review by udinobi — APPROVED** [review]

**Pull request merged**